src/prefect/engine/result/base.py

```diff
@@ -128,19 +116,20 @@ def store_safe_value(self) -> None:
                 value=value, result_handler=self.result_handler
             )
 
-    def populate_result(self, result: "Result") -> "Result":
+    def from_value(self, value: Any) -> "Result":
```

Comment:
@cicdw Is the change here because in the pipeline you are calling `state._result.read()` inline and then `self.task.result.from_value(what_you_read)`? and/or just generally if you can let me know the new thinking behind this name/signature change

Reply:
Yup yup for sure @lauralorenz ; so there are two patterns we need within the pipeline:

- creating result instances on the fly from values; this is what `from_value` method is for.  This way we can share the configuration that comes from initialization across different pieces of data.  We need to do this mainly for mapped tasks and looped tasks.
- "hydrating" result objects: this is necessary whenever we receive a State from Cloud which is only a JSON representation.  We need to use the data from that representation (specifically the `location`) and read from that location using the actual Result object.  This is what `populate_result` was for -- I realized in working through the pipeline that this felt more like an interface that _State_ objects should be responsible for, so in the other PR you'll notice this method has moved there.